### PR TITLE
docs: add API docs

### PR DESCRIPTION
**Pull request by andrewrynhard**
# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2249)

### CODE REVIEW

File: docs/website/content/v0.6.en.json [hunk 1]
```diff
@@ -98,6 +98,40 @@
...
+  {
+    "title": "API",
+    "path": "v0.6/en/api",
+    "items": [
+      {
+        "title": "Overview",
+        "path": "v0.6/en/api/common"
+      },
+      {
+        "title": "v1alpha1",
```
Comment (smira) [inline docs/website/content/v0.6.en.json:110]: should the title be different, like `health` ?

File: Dockerfile [hunk 2]
```diff
@@ -515,6 +515,17 @@ COPY --from=talosctl-linux /talosctl-linux-amd64 /bin/talosctl
...
+RUN protoc -I/protos -I/protos/common -I/protos/health --doc_out=/tmp --doc_opt=markdown,health.md /protos/health/*.proto
+RUN protoc -I/protos -I/protos/common -I/protos/machine --doc_out=/tmp --doc_opt=markdown,machine.md /protos/machine/*.proto
+RUN protoc -I/protos -I/protos/common -I/protos/network --doc_out=/tmp --doc_opt=markdown,network.md /protos/network/*.proto
+RUN protoc -I/protos -I/protos/common -I/protos/os --doc_out=/tmp --doc_opt=markdown,os.md /protos/os/*.proto
+RUN protoc -I/protos -I/protos/common -I/protos/security --doc_out=/tmp --doc_opt=markdown,security.md /protos/security/*.proto
+RUN protoc -I/protos -I/protos/common -I/protos/time --doc_out=/tmp --doc_opt=markdown,time.md /protos/time/*.proto
+
 FROM scratch AS docs
 COPY --from=docs-build /tmp/v1alpha1.md /docs/website/content/v0.6/en/configuration/v1alpha1.md
+COPY --from=proto-docs-build /tmp/*.md /docs/website/content/v0.6/en/api/common.md
```
Comment (smira) [inline Dockerfile:530]: should it be copy to `.../api` ? (without `common.md`?)

Comment (andrewrynhard) [reply Dockerfile:530]: :+1:

**Pull request closed without merging**